chat-server/src/routes/conversations.ts

```diff
@@ -50,11 +50,22 @@ conversationsRouter.post(
     try {
       // TODO: implement type checking on the request
 
+      const ipAddress = ""; // TODO: refactor to get IP address with middleware
+
       const stream = Boolean(req.params.stream);
       const { conversation, id } = req.body;
       const latestMessage = conversation[conversation.length - 1];
-      const embedding = await embeddings.createEmbedding(latestMessage.content);
-      const chunks = await database.content.findVectorMatches({ embedding });
+      const { status, embeddings: embeddingRes } =
+        await embeddings.createEmbedding({
+          text: latestMessage.content,
+          userIp: ipAddress,
+        });
+      if (status !== 200) {
+        return res.status(status).json({ error: "Embedding error" });
+      }
+      const chunks = await database.content.findVectorMatches({
+        embedding: embeddingRes!,
```

Comment:
The non-null assertion operator (`!`) doesn't really validate at runtime that embeddingRes is not null or undefined. It just tells the compiler that you are _sure_ it is defined. Since we aren't actually sure, you should add a case for it (like the non-200 status case above) or even a proper `assert()` if embeddingRes being null would be a programmer error.

Reply:
ditto https://github.com/mongodb/docs-chatbot/pull/6/files#r1229943342